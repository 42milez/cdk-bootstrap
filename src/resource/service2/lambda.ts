import { Construct, Duration } from '@aws-cdk/core';
import { Role } from '@aws-cdk/aws-iam';
import { AssetCode, Function, LayerVersion, Runtime, Tracing } from '@aws-cdk/aws-lambda';

export function createFunctions (scope: Construct, id: string, roles: { [key: string]: Role }, layers: { [key: string]: LayerVersion }): { [key: string]: Function } {
  return {
    hello: new Function(scope, 'HelloCdkFunction', {
      code: AssetCode.fromAsset('./src/function/hello'),
      description: 'Hello CDK',
      functionName: `hello-${id}`,
      handler: 'index.handler',
      memorySize: 128,
      role: roles.lambdaRole,
      runtime: Runtime.NODEJS_12_X,
      timeout: Duration.seconds(15),
      tracing: Tracing.ACTIVE
      // allowAllOutbound: -
      // deadLetterQueue: -
      // deadLetterQueueEnabled: -
      // environment: -
      // events: -
      // initialPolicy: -
      // logRetention: -
      // logRetentionRole: -
      // reservedConcurrentExecutions: -
      // securityGroup: -
      // securityGroups: -
      // vpc: -
      // vpcSubnets: -
    })
  };
}
