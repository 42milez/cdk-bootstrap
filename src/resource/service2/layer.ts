import { Construct } from '@aws-cdk/core';
import { Code, LayerVersion, Runtime } from '@aws-cdk/aws-lambda';

export function createLayers (scope: Construct, id: string, env: string, appName: string): { [key: string]: LayerVersion } {
  return {
    baseLayer1: new LayerVersion(scope, 'BaseLayer1', {
      compatibleRuntimes: [
        Runtime.NODEJS_12_X,
      ],
      code: Code.fromAsset(`layer.out/${env}/base1`),
      description: 'Base Layer 1',
      layerVersionName: `base-layer-1-${appName}-${env}`,
      // license: -
    }),
    baseLayer2: new LayerVersion(scope, 'BaseLayer2', {
      compatibleRuntimes: [
        Runtime.NODEJS_12_X,
      ],
      code: Code.fromAsset(`layer.out/${env}/base2`),
      description: 'Base Layer 2',
      layerVersionName: `base-layer-2-${appName}-${env}`,
      // license: -
    }),
  };
}
