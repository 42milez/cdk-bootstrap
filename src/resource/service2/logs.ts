import { Construct } from '@aws-cdk/core';
import { Function } from '@aws-cdk/aws-lambda';
import { LogGroup } from '@aws-cdk/aws-logs';

import { getRetentionDays } from '../../../lib/cdk/logs_helper';

export function createLogs (scope: Construct, functions: { [key: string]: Function }) {
  return {
    helloCdkFunctionLogGroup: new LogGroup(scope, 'HelloCdkFunctionLogGroup', {
      logGroupName: '/aws/function/' + functions.hello.functionName,
      retention: getRetentionDays()
    })
  };
}
