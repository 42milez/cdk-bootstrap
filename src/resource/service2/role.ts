import { Construct, Stack } from '@aws-cdk/core';
import { Effect, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from '@aws-cdk/aws-iam';

export function createRoles (scope: Construct, id: string): { [key: string]: Role } {
  return {
    lambdaRole: new Role(scope, 'LambdaRole', {
      assumedBy: new ServicePrincipal('function.amazonaws.com', {
        // conditions: -
        // region: -
      }),
      description: 'Hello CDK',
      inlinePolicies: {
        logs: new PolicyDocument({
          // assignSids: -
          statements: [
            new PolicyStatement({
              actions: [
                'logs:CreateLogGroup',
                'logs:CreateLogStream',
                'logs:PutLogEvents'
              ],
              effect: Effect.ALLOW,
              resources: [
                `arn:aws:logs:${Stack.of(scope).region}:${Stack.of(scope).account}:*`
              ]
              // conditions: -
              // notActions: -
              // notResources: -
              // notPrincipals: -
              // principals: -
              // sid: -
            })
          ]
        })
      },
      roleName: `lambda-${id}`
      // externalId: -
      // externalIds: -
      // managedPolicies: -
      // maxSessionDuration: -
      // path: -
      // permissionsBoundary: -
    })
  };
}
