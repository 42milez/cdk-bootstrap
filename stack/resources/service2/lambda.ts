import { Construct, Duration } from '@aws-cdk/core';
import { Role } from '@aws-cdk/aws-iam';
import { AssetCode, Function, LayerVersion, Runtime, Tracing } from '@aws-cdk/aws-lambda';

export function createFunctions (scope: Construct, roles: {[key:string]: Role}, layers: {[key:string]: LayerVersion}): {[key:string]: Function} {
    return {
        helloWorld: new Function(scope, 'HelloWorldFunction', {
            code: AssetCode.fromAsset('./src/hello-world'),
            description: 'Hello World',
            functionName: `hello-world-${process.env.APP_NAME}-${process.env.ENV}`,
            handler: 'index.handler',
            memorySize: 128,
            role: roles.lambdaRole,
            runtime: Runtime.NODEJS_12_X,
            timeout: Duration.seconds(15),
            tracing: Tracing.ACTIVE
            // allowAllOutbound: -
            // deadLetterQueue: -
            // deadLetterQueueEnabled: -
            // environment: -
            // events: -
            // initialPolicy: -
            // logRetention: -
            // logRetentionRole: -
            // reservedConcurrentExecutions: -
            // securityGroup: -
            // securityGroups: -
            // vpc: -
            // vpcSubnets: -
        })
    };
}
