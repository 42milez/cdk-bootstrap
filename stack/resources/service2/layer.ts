import { Construct } from '@aws-cdk/core';
import { Code, LayerVersion, Runtime } from '@aws-cdk/aws-lambda';

export function createLayers (scope: Construct): {[key:string]: LayerVersion} {
    return {
        baseLayer: new LayerVersion(scope, 'BaseLayer', {
            compatibleRuntimes: [
                Runtime.NODEJS_12_X
            ],
            code: Code.fromAsset(`layer.out/${process.env.ENV}`),
            description: 'Base Layer',
            layerVersionName: `base-layer-${process.env.APP_NAME}-${process.env.ENV}`
            // license: -
        })
    };
}
