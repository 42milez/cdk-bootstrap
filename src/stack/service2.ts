import * as cdk from '@aws-cdk/core';

import { createFunctions } from '../resource/service2/lambda';
import { createLayers } from '../resource/service2/layer';
import { createLogs } from '../resource/service2/logs';
import { createRoles } from '../resource/service2/role';

export class Service2Stack extends cdk.Stack {
  constructor (scope: cdk.Construct, id: string, env: string, appName: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // IAM Roles
    const roles = createRoles(this, id);

    // Lambda Layers
    const layers = createLayers(this, id, env, appName);

    // Lambda Functions
    const functions = createFunctions(this, id, env, roles, layers);

    // Log Groups
    createLogs(this, functions);
  }
}
