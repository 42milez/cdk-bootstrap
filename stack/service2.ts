import * as cdk from '@aws-cdk/core';

import { createFunctions } from './resources/service2/lambda';
import { createLayers } from './resources/service2/layer';
import { createLogs } from './resources/service2/logs';
import { createRoles } from './resources/service2/role';

export class Service2Stack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, env: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // IAM Roles
    const roles = createRoles(this, id);

    // Lambda Layers
    const layers = createLayers(this, id, env);

    // Lambda Functions
    const functions = createFunctions(this, id, roles, layers);

    // Log Groups
    createLogs(this, functions);
  }
}
