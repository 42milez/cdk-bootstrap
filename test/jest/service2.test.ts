import { SynthUtils } from '@aws-cdk/assert';
import { App } from '@aws-cdk/core';

import { Service2Stack } from '../../src/stack/service2';

test('Service2 Snapshot Test', () => {
  const app = new App();
  const stack = new Service2Stack(app, 'service2-development', 'development', 'cdk-bootstrap');
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
