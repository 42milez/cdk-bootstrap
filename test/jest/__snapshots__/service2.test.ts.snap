// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service2 Snapshot Test 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527dArtifactHashB08DD990": Object {
      "Description": "Artifact hash for asset \\"5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527d\\"",
      "Type": "String",
    },
    "AssetParameters5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527dS3BucketEAC44060": Object {
      "Description": "S3 bucket for asset \\"5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527d\\"",
      "Type": "String",
    },
    "AssetParameters5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527dS3VersionKeyE9F3085D": Object {
      "Description": "S3 key for asset version \\"5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527d\\"",
      "Type": "String",
    },
    "AssetParametersb6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605ArtifactHash685C7EC3": Object {
      "Description": "Artifact hash for asset \\"b6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605\\"",
      "Type": "String",
    },
    "AssetParametersb6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605S3Bucket3C7ED0F5": Object {
      "Description": "S3 bucket for asset \\"b6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605\\"",
      "Type": "String",
    },
    "AssetParametersb6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605S3VersionKey6BBDD3B1": Object {
      "Description": "S3 key for asset version \\"b6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605\\"",
      "Type": "String",
    },
    "AssetParameterse392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3ArtifactHashC7C87D0B": Object {
      "Description": "Artifact hash for asset \\"e392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3\\"",
      "Type": "String",
    },
    "AssetParameterse392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3S3Bucket40EA36AE": Object {
      "Description": "S3 bucket for asset \\"e392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3\\"",
      "Type": "String",
    },
    "AssetParameterse392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3S3VersionKey31154B55": Object {
      "Description": "S3 key for asset version \\"e392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "BaseLayer1D573D760": Object {
      "Properties": Object {
        "CompatibleRuntimes": Array [
          "nodejs12.x",
        ],
        "Content": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersb6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605S3Bucket3C7ED0F5",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersb6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605S3VersionKey6BBDD3B1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersb6b4159901c4c58ae071d371127d68a28c9e264a3583c7b77013c805a8917605S3VersionKey6BBDD3B1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Description": "Base Layer 1",
        "LayerName": "base-layer-1-cdk-experiment-development",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "BaseLayer2EEC4AED0": Object {
      "Properties": Object {
        "CompatibleRuntimes": Array [
          "nodejs12.x",
        ],
        "Content": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameterse392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3S3Bucket40EA36AE",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3S3VersionKey31154B55",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse392921aca9408c62afcbbebef20da8635c2c1fbc99fe88aeb87d991a66a17a3S3VersionKey31154B55",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Description": "Base Layer 2",
        "LayerName": "base-layer-2-cdk-experiment-development",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "HelloCdkFunction90393412": Object {
      "DependsOn": Array [
        "LambdaRoleDefaultPolicy75625A82",
        "LambdaRole3A44B857",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527dS3BucketEAC44060",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527dS3VersionKeyE9F3085D",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters5594105a5910777c67f1a01ff5f22fd414ca7ba0ad69928df6625ee2c9d0527dS3VersionKeyE9F3085D",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Description": "Hello CDK",
        "Environment": Object {
          "Variables": Object {
            "NODE_ENV": "development",
          },
        },
        "FunctionName": "hello-service2-development",
        "Handler": "index.handler",
        "Layers": Array [
          Object {
            "Ref": "BaseLayer1D573D760",
          },
          Object {
            "Ref": "BaseLayer2EEC4AED0",
          },
        ],
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "LambdaRole3A44B857",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 15,
        "TracingConfig": Object {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "HelloCdkFunctionLogGroup42EB1465": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/function/",
              Object {
                "Ref": "HelloCdkFunction90393412",
              },
            ],
          ],
        },
        "RetentionInDays": 1,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "LambdaRole3A44B857": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Description": "Hello CDK",
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "logs:CreateLogGroup",
                    "logs:CreateLogStream",
                    "logs:PutLogEvents",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:aws:logs:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "logs",
          },
        ],
        "RoleName": "lambda-service2-development",
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaRoleDefaultPolicy75625A82": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaRoleDefaultPolicy75625A82",
        "Roles": Array [
          Object {
            "Ref": "LambdaRole3A44B857",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
